adafruit_pyportal.py

```diff
@@ -1,952 +0,0 @@
-# SPDX-FileCopyrightText: 2017 Scott Shawcroft, written for Adafruit Industries
-# SPDX-FileCopyrightText: Copyright (c) 2020 Melissa LeBlanc-Williams for Adafruit Industries
-#
-# SPDX-License-Identifier: MIT
-"""
-`adafruit_pyportal`
-================================================================================
-
-A port of the PyPortal library intended to run on Blinka in CPython.
-
-* Author(s): Melissa LeBlanc-Williams
-
-Implementation Notes
---------------------
-
-**Software and Dependencies:**
-
-* Adafruit CircuitPython firmware for the supported boards:
-  https://github.com/adafruit/circuitpython/releases
-
-"""
-
-# imports
-import os
-import time
-import gc
-import subprocess
-import requests
-
-try:
-    import board
-
-    DISPLAY_ARG_REQUIRED = False
-except NotImplementedError:
```

Comment:
The way this was excepting this error was allowing it to work in environments where `board` doesn't exist. (a bit unusual I know.)

In particular it allows the library to work with my Blinka_Displayio_PyGame_Display driver, which passes a `display` argument that doesn't need any board pins or other devices to use.

Can we add back the logic and exception handling needed to support environments without `board`? I'm can work on that and add commits on this PR if you're okay with the change bout would rather not work on it specifically. Let me know.

Reply:
Could you please be more specific about what is missing. As far as I can tell, the logic is there, but was moved to the graphics module. Please keep in mind what PortalBase does as well as the fact that 3 other libraries also use PortalBase.

Reply:
Oh, I think I see what you're getting at. I changed the module to a package, so adafruit_pyportal.py was completely removed and the __init__.py file inside of the adafruit_pyportal folder is loaded instead. I did the same type of change in Adafruit_CircuitPython_PyPortal recently. No code changes are required. The logic is still there, but was moved to the appropriate place.

Reply:
I just tested with Blinka_Displayio_PyGameDisplay (as it is here) and it runs just fine.

Reply:
I get an error like this when I run it with the PyGame Display:
```
Traceback (most recent call last):
  File "/home/timc/repos/circuitpython/Blinka_Displayio_PyGameDisplay/examples/blinka_displayio_pygamedisplay_pyportal_bitcoin.py", line 13, in <module>
    from adafruit_pyportal import PyPortal
  File "/home/timc/repos/circuitpython/Blinka_Displayio_PyGameDisplay/examples/adafruit_pyportal/__init__.py", line 25, in <module>
    import board
  File "/home/timc/.local/lib/python3.8/site-packages/board.py", line 126, in <module>
    elif board_id == ap_board.CORAL_EDGE_TPU_DEV_MINI:
AttributeError: module 'adafruit_platformdetect.constants.boards' has no attribute 'CORAL_EDGE_TPU_DEV_MINI'
```
essentially anything that imports `board` would need try/except since my platform is not actually supported and cannot successfully import `board`. But I also don't need it since I'm passing the display in already prepared.

Reply:
Ok, I understand now. I pushed changes to account for this and tested by renaming my `board.py` file and it ran fine.